src/Equinor.ProCoSys.IPO.WebApi/Synchronization/StringToLongConverter.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Equinor.ProCoSys.IPO.WebApi.Synchronization;
+
+public class StringToLongConverter : JsonConverter<long>
+{
+    public override long Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType == JsonTokenType.String && long.TryParse(reader.GetString(), out long result))
+        {
+            return result;
+        }
+        throw new JsonException($"Unable to deserialize JSON to CommPkgEvent. Cannot convert {reader.GetString()} to {typeof(long)}.");
```

Comment:
This is a generic class which can be applied to other properties, so shouldn't specifically mention CommPkgEvent?
Although I appreciate trying to add context to the error message, is there another way to output the context?